src/commands/createFunctionApp/stacks/getStackPicks.ts

```diff
@@ -61,6 +62,14 @@ export async function getStackPicks(context: IFunctionAppWizardContext): Promise
                         break;
                 }
 
+                const endOfLifeDate = minorVersion.stackSettings.linuxRuntimeSettings?.endOfLifeDate;
+                if (endOfLifeDate) {
```

Comment:
I would make `shouldShowEolWarning` take `minorVersion` as an arg, so you can move these lines inside the function.